templates/commands/cel.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/abcxyz/abc/templates/model"
+	"github.com/abcxyz/pkg/logging"
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/common/types"
+	"github.com/google/cel-go/common/types/ref"
+)
+
+var (
+	celRegistry = types.NewEmptyRegistry()
+
+	celFuncs = []cel.EnvOption{
+		// Adds a split method on strings. Example: "foo,bar".split(",")
+		//
+		// Design decision: use a method instead of a plain function, because
+		// that is the convention that the community seems to prefer for string
+		// splitting. This is based on some quick googling of other peoples'
+		// custom split functions.
+		cel.Function(
+			"split",
+			cel.MemberOverload(
+				"string_split",
+				[]*cel.Type{cel.StringType, cel.StringType},
+				types.NewListType(types.StringType),
+				cel.BinaryBinding(func(lhs, rhs ref.Val) ref.Val {
+					toSplit := lhs.Value().(string)
```

Comment:
The linter is sad on these type assertions, but it's unclear if there's a way to return an error from this function.

Reply:
It's actually possible, the return value is a kind of implicit "Maybe" type. Fixed.